sdk/identity/identity/review/identity.api.md

```diff
@@ -106,26 +117,33 @@ export interface IdentityClientOptions extends ServiceClientOptions {
 
 // @public
 export class InteractiveBrowserCredential implements TokenCredential {
-    constructor(tenantId: string, clientId: string, options?: InteractiveBrowserCredentialOptions);
+    constructor(options?: InteractiveBrowserCredentialOptions);
     getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
 }
 
 // @public
 export interface InteractiveBrowserCredentialOptions extends IdentityClientOptions {
+    clientId?: string;
     loginStyle?: BrowserLoginStyle;
     postLogoutRedirectUri?: string | (() => string);
     redirectUri?: string | (() => string);
+    tenantId?: string;
 }
 
 // @public
 export const logger: import("@azure/logger").AzureLogger;
 
 // @public
 export class ManagedIdentityCredential implements TokenCredential {
-    constructor(clientId?: string, options?: IdentityClientOptions);
+    constructor(options?: ManagedIdentityCredentialOptions);
     getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
     }
 
+// @public
+export interface ManagedIdentityCredentialOptions extends IdentityClientOptions {
```

Comment:
Discussed offline with @bterlson and @daviwil
In this case, we want a constructor overload such that one constructor takes the clientId, the other doesnt.
Therefore, we can remove this new interface `ManagedIdentityCredentialOptions` and only add the overload

Reply:
Done.